app/src/main/java/com/github/lookupgroup27/lookup/ui/feed/Feed.kt

```diff
@@ -87,101 +96,142 @@ fun FeedScreen(
 
-                        postsViewModel.updatePost(
-                            post.copy(
-                                averageStars = newAvg,
-                                starsCount = newStarsCount,
-                                usersNumber = newUsersNumber,
-                                ratedBy =
-                                    if (!isReturningUser) {
-                                      post.ratedBy + userEmail
+                                val updatedRatings = profile?.ratings?.toMutableMap()
+                                updatedRatings?.set(post.uid, starsCount)
+                                val newProfile: UserProfile =
+                                    profile?.copy(
+                                        username = username,
+                                        bio = bio,
+                                        email = email,
+                                        ratings = updatedRatings ?: emptyMap())
+                                        ?: UserProfile(
+                                            username = username,
+                                            bio = bio,
+                                            email = email,
+                                            ratings = updatedRatings ?: emptyMap())
+                                profileViewModel.updateUserProfile(newProfile)
+
+                                val isReturningUser = post.ratedBy.contains(userEmail)
+                                val newStarsCount =
+                                    if (isReturningUser) {
+                                      post.starsCount - oldStarCounts + starsCount
                                     } else {
-                                      post.ratedBy
-                                    }))
-                      })
-                  Spacer(modifier = Modifier.height(20.dp))
+                                      post.starsCount + starsCount
+                                    }
+                                val newUsersNumber =
+                                    if (isReturningUser) post.usersNumber else post.usersNumber + 1
+                                val newAvg = newStarsCount.toDouble() / newUsersNumber
+
+                                postsViewModel.updatePost(
+                                    post.copy(
+                                        averageStars = newAvg,
+                                        starsCount = newStarsCount,
+                                        usersNumber = newUsersNumber,
+                                        ratedBy =
+                                            if (!isReturningUser) {
+                                              post.ratedBy + userEmail
+                                            } else {
+                                              post.ratedBy
+                                            }))
+                              })
```

Comment:
The logic for updating stars and user numbers can be encapsulated for clarity.
Suggested code:
```
fun calculatePostUpdates(
    post: Post,
    userEmail: String,
    starsCount: Int,
    oldStarCounts: Int
): Post {
    val isReturningUser = post.ratedBy.contains(userEmail)
    val newStarsCount = if (isReturningUser) {
        post.starsCount - oldStarCounts + starsCount
    } else {
        post.starsCount + starsCount
    }
    val newUsersNumber = if (isReturningUser) post.usersNumber else post.usersNumber + 1
    val newAvg = newStarsCount.toDouble() / newUsersNumber

    return post.copy(
        averageStars = newAvg,
        starsCount = newStarsCount,
        usersNumber = newUsersNumber,
        ratedBy = if (!isReturningUser) post.ratedBy + userEmail else post.ratedBy
    )
}

// Usage:
val updatedPost = calculatePostUpdates(post, userEmail, starsCount, oldStarCounts)
postsViewModel.updatePost(updatedPost)

```